java/org/contikios/cooja/plugins/BufferListener.java

```diff
@@ -694,20 +692,15 @@ public void actionPerformed(ActionEvent e) {
     });
 
     logUpdateAggregator.start();
-    simulation.getEventCentral().addMoteCountListener(logOutputListener = new MoteCountListener() {
-      @Override
-      public void moteWasAdded(Mote mote) {
-        /* Update title */
-        try {
-          startMonitoring(mote);
-        } catch (Exception e) {
-          logger.warn("Could not monitor buffer on: " + mote, e);
+    simulation.getMoteTriggers().addTrigger(this, (event, m) -> {
+      try {
+        if (event == EventTriggers.AddRemove.ADD) {
+          startMonitoring(m);
+        } else {
+          stopObserving(m);
         }
-      }
-      @Override
-      public void moteWasRemoved(Mote mote) {
-        /* Update title */
-        stopObserving(mote);
+      } catch (Exception e) {
+        logger.warn("Could not monitor buffer on: " + m, e);
```

Comment:
It looks like only `startMonitoring(m)` throws exceptions. What do you think about moving in the try-catch around only the call to `startMonitoring(m)`?

Reply:
Fixed.